gallery/plot_datapoints.py

```diff
@@ -0,0 +1,122 @@
+
+image = datapoints.Image([[0, 1], [1, 0]])
+print(image)
+
+
+########################################################################################################################
+# Similar to other PyTorch creations ops, the constructor also takes the ``dtype``, ``device``, and ``requires_grad``
+# parameters.
+
+float_image = datapoints.Image([[0, 1], [1, 0]], dtype=torch.float32, requires_grad=True)
+print(float_image)
+
+
+########################################################################################################################
+# In addition, :class:`~torchvision.datapoints.Image` and :class:`~torchvision.datapoints.Mask` also take a
+# :class:`PIL.Image.Image` directly:
+
+image = datapoints.Image(PIL.Image.open("assets/astronaut.jpg"))
+print(image.shape, image.dtype)
+
+########################################################################################################################
+# In general, the datapoints can also store additional metadata that complements the underlying tensor. For example,
+# :class:`~torchvision.datapoints.BoundingBox` stores the format as well as the spatial size of the corresponding image
+# alongside the actual values:
+
+bounding_box = datapoints.BoundingBox(
+    [17, 16, 344, 495], format=datapoints.BoundingBoxFormat.XYXY, spatial_size=image.shape[-2:]
+)
+print(bounding_box)
+
+
+########################################################################################################################
+# Do I have to wrap the output of the datasets myself?
+# ----------------------------------------------------
+#
+# Only if you are using custom datasets. For the builtin ones, you can use
+# :func:`torchvision.datasets.wrap_dataset_for_transforms_v2`. Note that the function also supports subclasses of the
+# builtin datasets. Meaning, if your custom dataset subclasses from a builtin one and the output type is the same, you
+# also don't have to wrap manually.
+#
+# How do the datapoints behave inside a computation?
+# --------------------------------------------------
+#
+# Datapoints look and feel just like regular tensors. Everything that is supported on plain :class:`torch.Tensor`'s
+# also works on datapoints.
+# Since for most operations involving datapoints, it cannot be safely inferred whether the result should retain the
+# datapoint type, the result will be a plain tensor:
+
+assert type(image) is datapoints.Image
+
+new_image = image + 0
+
+assert type(new_image) is torch.Tensor
+
+########################################################################################################################
+# There are two exceptions to this rule:
+#
+# 1. The operations :meth:`~torch.Tensor.clone`, :meth:`~torch.Tensor.to`, and :meth:`~torch.Tensor.requires_grad_`
+#    retain the datapoint type.
+# 2. Inplace operations on datapoints cannot change the type of the datapoint they are called on. However, if you use
+#    the flow style, the returned value will be unwrapped:
```

Comment:
What is the flow style?

Reply:
`foo.bar_().baz_()`. Meaning, although the operation applies inplace, we still return something to allow chaining multiple ops together. I've seen the term "flow style" or "flowing style" for that. Happy to use a different one.

Reply:
got it, perhaps I would have guessed the meaning if there had been more than one chained call, but not sure.

Reply:
Yeah, let's add one more to hammer this home